docs/source/async_api.rst

```diff
@@ -163,29 +163,41 @@ For example:
 
 .. code-block:: python
 
-    from neo4j import AsyncGraphDatabase
+   import neo4j
 
 
-    async def custom_resolver(socket_address):
-        if socket_address == ("example.com", 9999):
-            yield "::1", 7687
-            yield "127.0.0.1", 7687
-        else:
-            from socket import gaierror
-            raise gaierror("Unexpected socket address %r" % socket_address)
+    def custom_resolver(socket_address):
+        # assert isinstance(socket_address, neo4j.Address)
+        if socket_address != ("example.com", 9999):
+            raise OSError(f"Unexpected socket address {socket_address!r}")
+
+        # You can return any neo4j.Address object
+        yield neo4j.Address(("localhost", 7687))  # IPv4
+        yield neo4j.Address(("::1", 7687, 0, 0))  # IPv6
+        yield neo4j.Address.parse("localhost:7687")
+        yield neo4j.Address.parse("[::1]:7687")
+
+        # or any tuple that can be passed to neo4j.Address(...).
+        # Initially, this will be interpreted as IPv4, but DNS resolution
+        # will turn it into IPv6 if appropriate.
+        yield "::1", 7687
+        # This will be interpreted as IPv6 directly, but DNS resolution will
+        # happen still.
+        yield "::1", 7687, 0, 0
```

Comment:
One question this example raises for me is why I should use this way, when I can just do `yield "::1", 7687` and it has the same effect? Is there any advantage in being explicit with 3rd and 4th entries?

Reply:
Quoting the Python docs

> For [socket](https://docs.python.org/3/library/socket.html#module-socket) module methods, flowinfo and scope_id can be omitted just for backward compatibility. Note, however, omission of scope_id can cause problems in manipulating scoped IPv6 addresses.

So while I can't come up with an example, because frankly I don't know enough about IPv6, there might be.